### PR TITLE
fix(collectd): increase collectd tmpfs based of disks and CPUs

### PR DESCRIPTION
**Pull request by william-gr**
Also stop collecting geom_stat for every partition.

Ticket:	#28792

### CODE REVIEW

File: src/freenas/etc/ix.rc.d/ix-collectd [hunk 1]
```diff
@@ -126,7 +126,13 @@ generate_collectd()
...
         if [ $datadir_is_ramdisk -eq 0 ];
         then
-            mount -t tmpfs -o size=600m tmpfs "${datadir}"
+            # Each disk consumes about 1.7MB (geom_stat) + 1.1MB(disk) of space
+            # so we need bigger tmpfs on systems with lots of disks.
+            no_disks=$(sysctl -n kern.disks | tr -s " " "\n" | wc -l)
+            # Each CPU takes about 1MB
+            no_cpus=$(sysctl -n kern.smp.cpus)
+            size=$(echo "(300 + ${no_disks}*2.8 + ${no_cpus}*1) / 1" | bc)
+            mount -t tmpfs -o size=${size}m tmpfs "${datadir}"
```
Comment (amotin) [inline src/freenas/etc/ix.rc.d/ix-collectd]: Should this be so flexible?  It is tmpfs, not a dumb RAM disk of fixed size. I suppose it should not consume more space then needed.  I worry this logic may not work if JBOD is connected after the system is booted.

Comment (william-gr) [reply src/freenas/etc/ix.rc.d/ix-collectd]: @amotin  You're right, I will just fix itt to 2g, then?

Comment (themylogin) [reply src/freenas/etc/ix.rc.d/ix-collectd]: @william-gr if something bad happens, it will eat up 2GB of RAM then?

Comment (amotin) [reply src/freenas/etc/ix.rc.d/ix-collectd]: Would be fine for me.  Though the part above removing partition stats looks even better.  Spending gigabytes of RAM for stats is expensive.

Comment (william-gr) [reply src/freenas/etc/ix.rc.d/ix-collectd]: @themylogin yes, thats the downside

@amotin true, but we are not taking into account the datasets. Every dataset also adds to the collectd rrd overall size. So I am not sure 600m would be enough for everything.
However I do believe we reload collectd when we added/remove datasets. Is it possible to resize a tmpfs?
Or should we just set it to something more reasonable like 1g and be done with it?

Comment (amotin) [reply src/freenas/etc/ix.rc.d/ix-collectd]: I am not saying we should not increase the size, I am just happy that it is not the only way used.

**Pull request merged**